src/posts/2020-07-30-the-worst-decision-of-my-career.md

```diff
@@ -0,0 +1,271 @@
+---
+path: /posts/the-worst-decision-of-my-career/
+title: "The Worst Decision Of My Career"
+author: gabriel-poca
+date: 2020-07-30
+tags:
+  - development
+intro: "A complex system that works is invariably found to have evolved from a simple system that worked. A complex system designed from scratch never works and cannot be patched up to make it work. You have to start over with a working simple system"
+---
+
+This is a reflection on software development and complexity. I'll talk about
+tools, technology, architecture, planning, product development, and other
+things. Let's start with some quotes to make me look smart:
+
+> A complex system that works is invariably found to have evolved from a simple
+> system that worked. A complex system designed from scratch never works and
+> cannot be patched up to make it work. You have to start over with a working
+> simple system -- _Gall's law_
+
+I only came across this quote recently, but I think that it summarizes what
+I've learned these past seven years.
+
+> Programming in isolation is problem-solving, but software engineering is all
+> about managing complexity.
+
+This is an idea that can be found in many books, and the distinction between
+programming and software engineering makes sense to me, probably because of my
+education and mental models. I'm sure we can all disagree on this, but that's
+not the topic of today. Complexity is.
+
+## Complexity
+
+Almost everyone I know in the software industry wants to build products that
+solve real and hard problems. Change the world! If you're working on the web,
+the last thing you want is to build another CRUD application. They are fine,
+but it gets boring after a while.
+
+Most of us will get bored without novelty. That's why the software industry has
+a recycling mechanism to keep things fresh and interesting for us: every once
+in a while a new, or different, language/framework will rise and light the path
+for a brighter future, overthrowing the existing standard, demanding that we
+learn it to be on top of the game.
+
+We change our tools, but we keep building the same things over and over. I've
+been doing this long enough to see that we are just going in circles. Let's be
+honest, most of us aren't building life-changing products that wouldn't have
+been possible ten years ago, so let's not jump to the sight of another shiny
```

Comment:
I don't think "jump to the sight" is an expression. Maybe "jump straight into another"?

Reply:
👍